### PR TITLE
[Devfile] Remove namespace flag from odo commands

### PR DESCRIPTION
**Pull request by johnmcollier**
**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

/kind cleanup
/area devfile

**What does does this PR do / why we need it**:
As mentioned earlier, we decided to use only the `--project` flag in odo for devfile components and to not use `--namespace` at this moment. This PR removes the `--namespace` flag that was added to certain odo commands for use with devfile components and uses `--project` instead. For commands that didn't have the `--project` flag added to it, the flag was added. 

Integration tests updated to use `--project` only as well.

**Which issue(s) this PR fixes**:

Fixes https://github.com/openshift/odo/issues/2769

**How to test changes / Special notes to the reviewer**:
- Run the unit and integration tests
- `odo create --project <project>` -> Verify namespace/project in env.yaml is correct
- `odo push --project <project>` -> should push to the proper namespace
- `odo delete --project <project>` -> should delete from the proper namespace
- `odo watch --project <project>` -> should push changes to the proper namespace

### CODE REVIEW

**Comment by johnmcollier** [comment]
/retest 

More OpenShift CI flakiness

File: pkg/odo/cli/component/push.go [hunk 1]
```diff
@@ -65,6 +66,13 @@ func (po *PushOptions) Complete(name string, cmd *cobra.Command, args []string)
 		}
 		po.EnvSpecificInfo = envinfo
 		po.Context = genericclioptions.NewDevfileContext(cmd)
+
+		if cmd.Flags().Changed("project") {
```
Comment (mik-dass) [inline pkg/odo/cli/component/push.go:70]: I think this needs to be moved here https://github.com/openshift/odo/blob/6740e5177ae11db734f7d3d303c4f689015e3b91/pkg/odo/genericclioptions/context.go#L343. We are repeating the same steps in all the above files.

Comment (johnmcollier) [reply pkg/odo/cli/component/push.go:70]: @mik-dass You're right, good catch!

Comment (johnmcollier) [reply pkg/odo/cli/component/push.go:70]: Actually, we don't need to move any code to `resolveNamespace`, just update it to retrieve the namespace from the `--project` flag.

So long as we initialize the devfile context in each command that needs it, we can retrieve the namespace via `kclient.Namespace`

**Comment by johnmcollier** [comment]
More CI infra flakes 😢 

/retest

**Comment by mik-dass** [comment]
@johnmcollier needs rebase

**Comment by johnmcollier** [comment]
/retest

Giving the builds a kick due to the gosec issues earlier

**Comment by johnmcollier** [comment]
@mik-dass I've rebased and addressed your review comment (retrieving namespace from the getNamespace function)

**Comment by johnmcollier** [comment]
There may be a potential merge conflict with this and https://github.com/openshift/odo/issues/2922, so I'm holding this so that it doesn't slip in to this afternoon's release and delay 2922. I'll cancel the hold after the release.

/hold

**Comment by maysunfaisal** [comment]
tests were updated! Thx
/lgtm

**Pull request merged**